test/integration/builder/inserts.js

```diff
@@ -648,6 +648,122 @@ module.exports = function(knex) {
+            ['test7@example.com','Test','User', function (v) {return v.toString() === '[object ReturningHelper:id]';}],
+            [8]
+          );
+          tester(
+            'mssql',
+            'insert into [accounts] ([email], [first_name], [last_name]) output inserted.[id] values (?, ?, ?)',
+            ['test7@example.com','Test','User'],
+            ['8']
+          );
+        });
+
+    });
+
+    it('should allow insert on conflict do update in postgres', function() {
+
+      return knex('accounts')
+        .insert({
+          email:'test8@example.com',
+          first_name: 'User',
+          last_name: 'Test'
+        }, 'id')
+        .onConflict('email', {
+          first_name: 'Test',
+          last_name: 'User'
+        })
+        .testSql(function(tester) {
+          tester(
+            'mysql',
+            'insert into `accounts` (`email`, `first_name`, `last_name`) values (?, ?, ?)',
+            ['test8@example.com','Test','User'],
+            [9]
+          );
+          tester(
+            'postgresql',
+            'insert into "accounts" ("email", "first_name", "last_name") values (?, ?, ?) on conflict ("email") do update set "first_name" = \'Test\', "last_name" = \'User\' returning "id"',
+            ['test8@example.com','Test','User'],
+            ['9']
+          );
+          tester(
+            'sqlite3',
+            'insert into `accounts` (`email`, `first_name`, `last_name`) values (?, ?, ?)',
+            ['test8@example.com','Test','User'],
+            [8]
+          );
+          tester(
+            'oracle',
+            "insert into \"accounts\" (\"email\", \"first_name\", \"last_name\") values (?, ?, ?) returning ROWID into ?",
+            ['test8@example.com','Test','User', function (v) {return v.toString() === '[object ReturningHelper:id]';}],
+            [9]
+          );
+          tester(
+            'mssql',
+            'insert into [accounts] ([email], [first_name], [last_name]) output inserted.[id] values (?, ?, ?)',
+            ['test8@example.com','Test','User'],
+            ['9']
+          );
+        });
+
+    });
+
+    it('should perform update on conflict in postgres and return conflicting row', function() {
+      if (knex.client.dialect === 'postgresql') {
+        return knex('accounts')
+          .insert({
+            email:'test8@example.com',
+            first_name: 'User',
+            last_name: 'Test'
+          }, 'id')
+          .onConflict('email', {
+            first_name: 'Testing',
+            last_name: 'Users'
+          })
+          .testSql(function(tester) {
+            tester(
+              'postgresql',
+              'insert into "accounts" ("email", "first_name", "last_name") values (?, ?, ?) on conflict ("email") do update set "first_name" = \'Testing\', "last_name" = \'Users\' returning "id"',
+              ['test8@example.com','Test','User'],
+              ['9']
+            );
+          });
+      }
+    });
```

Comment:
In addition to unit tests which are just generating queries,  there should be real integration test, which actually tries to insert stuff to db and checks that in conflict situation there was correct data added to DB.